core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -1952,6 +1989,220 @@ void testDateFormat(DateFormatTester dft) {
+            "01:14:15 PM"),
+        Arguments.of(
+            DSL.literal("1998-01-31 13:14:15.012345"),
+            DSL.literal("%S"),
+            "15"),
+        Arguments.of(
+            DSL.literal("1998-01-31 13:14:15.012345"),
+            DSL.literal("%s"),
+            "15"),
+        Arguments.of(
+            DSL.literal("1998-01-31 13:14:15.012345"),
+            DSL.literal("%T"),
+            "13:14:15"),
+        Arguments.of(
+            DSL.literal("1998-01-31 13:14:15.012345"),
+            DSL.literal("%f %H %h %I %i %k %l %p %r %S %s %T"),
+            "012345 13 01 01 14 13 1 PM 01:14:15 PM 15 15 13:14:15")
+    );
+  }
+
+  private void timeFormatQuery(LiteralExpression arg,
+                               LiteralExpression format,
+                               String expectedResult) {
+    FunctionExpression expr = DSL.time_format(functionProperties, arg, format);
+    assertEquals(STRING, expr.type());
+    assertEquals(expectedResult, eval(expr).stringValue());
+  }
+
+  @ParameterizedTest(name = "{0}{1}")
+  @MethodSource("getTestDataForTimeFormat")
+  public void testTimeFormat(LiteralExpression arg,
+                             LiteralExpression format,
+                             String expectedResult) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    timeFormatQuery(arg, format, expectedResult);
+  }
+
+  private static Stream<Arguments> getInvalidTestDataForTimeFormat() {
+    return Stream.of(
+        Arguments.of(
+            DSL.literal("asdfasdf"),
+            DSL.literal("%f")),
+        Arguments.of(
+            DSL.literal("12345"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("10:11:61"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("10:61:12"),
+            DSL.literal("%h")),
+        Arguments.of(
+            DSL.literal("61:11:12"),
+            DSL.literal("%h"))
+    );
+  }
+  
+  @ParameterizedTest(name = "{0}{1}")
+  @MethodSource("getInvalidTestDataForTimeFormat")
+  public void testInvalidTimeFormat(LiteralExpression arg, LiteralExpression format) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+    FunctionExpression expr = DSL.time_format(functionProperties, arg, format);
+    assertThrows(SemanticCheckException.class, () -> eval(expr));
+  }
+
+  private static Stream<Arguments> getInvalidTimeFormatHandlers() {
+    return Stream.of(
+        Arguments.of("%a"),
+        Arguments.of("%b"),
+        Arguments.of("%j"),
+        Arguments.of("%M"),
+        Arguments.of("%W"),
+        Arguments.of("%D"),
+        Arguments.of("%w"),
+        Arguments.of("%U"),
+        Arguments.of("%u"),
+        Arguments.of("%V"),
+        Arguments.of("%v"),
+        Arguments.of("%X"),
+        Arguments.of("%x")
```

Comment:
you could also test formats without `%` or with extra characters like `%%h` or `%hh` or wrong order or extra spaces like `h%` or ` %h`